components/zendesk/sources/solved-ticket/solved-ticket.mjs

```diff
@@ -0,0 +1,52 @@
+import common from "../common.mjs";
+
+export default {
+  ...common,
+  name: "Solved Ticket",
+  key: "zendesk-solved-ticket",
+  type: "source",
+  description: "Emit new event when a ticket is solved",
+  version: "0.0.1",
+  methods: {
+    ...common.methods,
+    getWebhookName() {
+      return "Solved Ticket Webhook";
+    },
+    getTriggerTitle() {
+      return "Solved Ticket Trigger";
+    },
+    getTriggerConditions() {
+      return {
+        any: [
+          {
+            field: "status",
+            operator: "is",
+            value: "solved",
+          },
+        ],
+      };
+    },
+    getTriggerPayload() {
+      return {
+        ticketId: "{{ticket.id}}",
+        title: "{{ticket.title}}",
+        description: "{{ticket.description}}",
+        url: "{{ticket.url}}",
+        requester: "{{ticket.requester.first_name}} {{ticket.requester.last_name}} <{{ticket.requester.email}}>",
+        assignee: "{{ticket.assignee.first_name}} {{ticket.assignee.last_name}} <{{ticket.assignee.email}}>",
+        status: "{{ticket.status}}",
+        createdAt: "{{ticket.created_at}}",
+        updatedAt: "{{ticket.updated_at}}",
+      };
+    },
+  },
+  async run(event) {
+    const payload = event.body;
+
+    this.$emit(payload, {
+      id: payload.ticketId,
```

Comment:
As mentioned in the comment above:
If an event is emitted each time a ticket's status is changed to "solved", this `id` property might not [uniquely identify events](https://pipedream.com/docs/components/api/#emit).

To fix this, the `id` could include the event's timestamp. For example:
```js
    const ts = Date.parse(payload.updatedAt);
    const id = `${payload.ticketId}-${ts}`;
    this.$emit(payload, {
      id,
      ts,
```